src/backend/marsha/core/tests/test_api_video.py

```diff
@@ -2654,6 +2726,91 @@ def test_api_video_patch_detail_token_user_is_public(self):
         video.refresh_from_db()
         self.assertTrue(video.is_public)
 
+    def test_api_video_patch_detail_token_user_allow_recording(self):
+        """Instructors and administrators should be able to
+        patch the recording flag of their video through the API."""
+        video = factories.VideoFactory()
+        self.assertTrue(video.allow_recording)
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+        data = {"allow_recording": False}
+        response = self.client.patch(
+            f"/api/videos/{video.id}/",
+            data,
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        video.refresh_from_db()
+        self.assertFalse(video.allow_recording)
+
+    def test_api_video_patch_detail_token_user_estimated_duration(self):
+        """Instructors and administrators should be able to
+        patch the estimated_duration field of their video through the API."""
+        video = factories.VideoFactory()
+        self.assertIsNone(video.estimated_duration)
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+        data = {"estimated_duration": "5400"}
```

Comment:
Can you also add a test with a negative integer please

Reply:
I added it.